x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -7,39 +7,66 @@
 import React from 'react';
 import styled from 'styled-components';
 
+import { EuiIcon } from '@elastic/eui';
 import {
   colors,
+  fontFamily,
   fontFamilyCode,
+  fontSize,
   fontSizes,
   px,
   unit,
   units
 } from '../../../../../../style/variables';
 import { IWaterfallItem } from './waterfall_helpers/waterfall_helpers';
 
-const ItemBar = styled.div`
+interface ItemBarProps {
+  type: 'transaction' | 'span';
+  left: number;
+  width: number;
+  color: string;
+}
+
+const ItemBar = styled<ItemBarProps, any>('div')`
+  box-sizing: border-box;
   position: relative;
-  height: ${unit}px;
+  height: ${props => (props.type === 'transaction' ? unit : unit)}px;
+  left: ${props => `${props.left}%`};
+  width: ${props => props.width}%;
```

Comment:
Use just noticed the different concat approaches above. Should we try to stick to one? :)